### ISSUE TITLE
v_error Memory Leak 

### ISSUE THREAD

**Issue by wilesun** [vlang/v#9384]
**V version:V 0.2.2 ead0dff
**OS:ubuntu 18.04

**What did you do?**
when call a function with return error, v_error's memdup memory is not free
```
fn test1()?{
	return error("this is a test")
}

fn main() {
	test1() or {
		println(err)
		return
	}
	println("ok")
}
```

**What did you expect to see?**

 
**What did you see instead?**
or catch the error,defer the or code then free err memory

**Comment by Larpon** [comment]
Is this compiled with `-autofree`?

**Comment by Delta456** [comment]
This is what I get with using `-autofree`:
```
==21904== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==21904== Using Valgrind-3.16.1 and LibVEX; rerun with -h for copyright info
==21904== Command: foo
==21904== 
this is a test
==21904== 
==21904== HEAP SUMMARY:
==21904==     in use at exit: 24 bytes in 1 blocks
==21904==   total heap usage: 21 allocs, 20 frees, 18,319 bytes allocated
==21904== 
==21904== LEAK SUMMARY:
==21904==    definitely lost: 24 bytes in 1 blocks
==21904==    indirectly lost: 0 bytes in 0 blocks
==21904==      possibly lost: 0 bytes in 0 blocks
==21904==    still reachable: 0 bytes in 0 blocks
==21904==         suppressed: 0 bytes in 0 blocks
==21904== Rerun with --leak-check=full to see details of leaked memory
==21904== 
==21904== For lists of detected and suppressed errors, rerun with: -s
==21904== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```
Without using `-autofree`:
```
==21924== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==21924== Using Valgrind-3.16.1 and LibVEX; rerun with -h for copyright info
==21924== Command: foo
==21924== 
this is a test
==21924== 
==21924== HEAP SUMMARY:
==21924==     in use at exit: 17,280 bytes in 19 blocks
==21924==   total heap usage: 20 allocs, 1 frees, 18,304 bytes allocated
==21924== 
==21924== LEAK SUMMARY:
==21924==    definitely lost: 24 bytes in 1 blocks
==21924==    indirectly lost: 0 bytes in 0 blocks
==21924==      possibly lost: 0 bytes in 0 blocks
==21924==    still reachable: 17,256 bytes in 18 blocks
==21924==         suppressed: 0 bytes in 0 blocks
==21924== Rerun with --leak-check=full to see details of leaked memory
==21924== 
==21924== For lists of detected and suppressed errors, rerun with: -s
==21924== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

**Comment by wilesun** [comment]
V version:V 0.2.2 5873149 ,compile with -autofree,but valgrind check memory leak,problem still is:
```
valgrind --leak-check=full --show-leak-kinds=all ./test2
==19333== Memcheck, a memory error detector
==19333== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==19333== Using Valgrind-3.16.1 and LibVEX; rerun with -h for copyright info
==19333== Command: ./test2
==19333== 
this is a test
==19333== 
==19333== HEAP SUMMARY:
==19333==     in use at exit: 24 bytes in 1 blocks
==19333==   total heap usage: 21 allocs, 20 frees, 18,319 bytes allocated
==19333== 
==19333== 24 bytes in 1 blocks are definitely lost in loss record 1 of 1
==19333==    at 0x4C31ECB: malloc (vg_replace_malloc.c:307)
==19333==    by 0x4155A5: v_malloc (in /test3/test2)
==19333==    by 0x415862: memdup (in /test3/test2)
==19333==    by 0x41A6B3: v_error (in /test3/test2)
==19333==    by 0x425B90: main__test1 (in /test3/test2)
==19333==    by 0x425C20: main__main (in /test3/test2)
==19333==    by 0x4381CE: main (in /media/test3/test2)
==19333== 
==19333== LEAK SUMMARY:
==19333==    definitely lost: 24 bytes in 1 blocks
==19333==    indirectly lost: 0 bytes in 0 blocks
==19333==      possibly lost: 0 bytes in 0 blocks
==19333==    still reachable: 0 bytes in 0 blocks
==19333==         suppressed: 0 bytes in 0 blocks
==19333== 
==19333== For lists of detected and suppressed errors, rerun with: -s
==19333== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
```

**Comment by medvednikov** [comment]
Fixed in 0.3
